### PR TITLE
added VALARM to birthday event

### PR DESCRIPTION
**Pull request by Garfonso**
This was discussed for the contacts app in the past but had a bit of trouble getting in. Now the code has been moved to core, so I'll try here again.

One result of the past discussion (can't find it anymore) was to set ACTION=DISPLAY, so it _should_ only display a notification and not make a sound.

It might be preferable to make this configurable. But I don't know how to achieve this, right now. 

Anyway, I put this up for discussion, again and would be happy to see it getting merged. Otherwise I'll have to live with the integrity warning, for now.

### CODE REVIEW

**Comment by RealRancor** [comment]
Hi,

great to see this again. Had considered to create a new feature request for this after "the dust have settled" for the oC9 release.

I think the main issue here is still, that the birthday calendar needs to make writable. Without that errors like seen in https://github.com/owncloudarchive/contacts/pull/440#issuecomment-162167784 will shown when dismissing a reminder.

**Comment by Garfonso** [comment]
@owncloud-bot ok, did send the form. Let's see what happens next. :)

@RealRancor: The issue with Thunderbird? Actually I'm more convinced that this is an issue in Thunderbird, not so much in owncloud. 
Because from what I read about ICS and CalDav there is no need anywhere to upload an appointment if an alarm is dismissed. And other calender implementations don't do that at all, for good reasons.

In fact Thunderbird gives me a lot of conflicts also with writable events. This is probably because I use Thunderbird on multiple computers and I get the notifications on all of them, but one of them is the first to update the event on the server and all others will get a mismatch in the etag value and throw me a conflict warning.
I now disabled alarms in Thunderbird completely (in addition to these issues, if there were many alarms to catch up, because TB was off for a few days, it was lagging a lot, often for a few minutes and processing all these alarms. Quite nasty) and those issues are gone. Of course, not the best solution... but calendar in TB always was a bit strange and I did not see any of this in other calendars (for example even the Windows 10 calendar app works quite nicely with these alarms, on mobile as on desktop).

**Comment by LukasReschke** [comment]
cc @DeepDiver1975 @georgehrke

File: apps/dav/lib/caldav/birthdayservice.php [hunk 1]
```diff
@@ -153,6 +153,11 @@ public function buildBirthdayFromContact($cardData) {
 		$vEvent->{'RRULE'} = 'FREQ=YEARLY';
 		$vEvent->{'SUMMARY'} = $title . ' (*' . $date->format('Y') . ')';
 		$vEvent->{'TRANSP'} = 'TRANSPARENT';
+		$alarm = $vCal->createComponent('VALARM');
+		$alarm->add($vCal->createProperty('TRIGGER', '-PT0M', ['VALUE' => 'DURATION']));
+		$alarm->add($vCal->createProperty('ACTION', 'DISPLAY'));
+		$alarm->add($vCal->createProperty('DESCRIPTION', $title . ' (' . $date->format('Y') . ')'));
```
Comment (DeepDiver1975) [inline apps/dav/lib/caldav/birthdayservice.php]: please reuse the string as in line 154 - THX

**Comment by RealRancor** [comment]
> The issue with Thunderbird? Actually I'm more convinced that this is an issue in Thunderbird, not so much in owncloud. 

Also seeing this together with DAVDroid 0.9.x . I think all clients are trying to write into an event when delaying or dismissing a reminder. If not you won't be able to dismiss/delay an alarm on all other synced devices.

**Comment by karlitschek** [comment]
confirmed that I got the CLA

**Comment by DeepDiver1975** [comment]
> confirmed that I got the CLA

THX @karlitschek 

@Garfonso mind re-opening this change in a branch within this repo? CI can then be executed easily - THX

**Comment by Garfonso** [comment]
@DeepDiver1975 I created a branch for this change here: https://github.com/owncloud/core/tree/alarms_for_birthdayevents

Should I do a new pull request from that branch to master?

**Comment by DeepDiver1975** [comment]
> Should I do a new pull request from that branch to master?

yes please - THX. I'm closing this PR in favor of the new one

**Pull request closed without merging**